refinery/data_set_manager/tests.py

```diff
@@ -1605,6 +1610,22 @@ def test_update_annotated_nodes(self):
         self.assertEqual(len(nodes_after), 0)
         # TODO: Is this the behavior we expect?
 
+    def test_update_existing_dataset_with_revised_investigation(self):
```

Comment:
why not break it into three unit tests, one for get_version, get_investigation, and get_latest_investigation_link. It was make debugging easier down the road.